sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp

```diff
@@ -457,12 +459,13 @@ __esimd_dpas_bits_precision(__ESIMD_ENS::argument_type precisionType) {
              : 1;
 }
 
+// TODO : 'src0' argument type other than 'void*'?
 template <__ESIMD_ENS::argument_type src1_precision,
           __ESIMD_ENS::argument_type src2_precision, int systolic_depth,
           int repeat_count, typename RT, typename T1, typename T2,
           __ESIMD_NS::uint SZ, __ESIMD_NS::uint N1, __ESIMD_NS::uint N2>
 inline __ESIMD_DNS::vector_type_t<RT, SZ>
-__esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,
+__esimd_dpas_inner(const void *src0,
```

Comment:
why src0 needs to be void* ?

Reply:
For following tests, `__ESIMD_DNS::vector_type_t<RT, SZ> *` is working fine.

```
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/dpas/dpas_test1.cpp
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/dpas/dpas_test2.cpp
https://github.com/intel/llvm-test-suite/blob/intel/SYCL/ESIMD/dpas/dpas_test3.cpp
```

However, following test fails.

```
https://github.com/intel/llvm/blob/sycl/sycl/test/esimd/dpas.cpp
```

Compilation error message : 
```
llvm_build/install/bin/../include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:669:29: error: static_cast from 'short * __attribute__((ext_vector_type(16)))' to 'sycl::ext::intel::esimd::detail::vector_type_t<floa
t, 16> *' (aka 'cl::sycl::ext::intel::esimd::detail::raw_vector_type<float, 16>::type *') is not allowed
                            static_cast<__ESIMD_DNS::vector_type_t<T, N> *>(std::addressof(src0)),
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

As I was not sure what type of argument could work, I used `void *` (with TODO comment)

Reply:
I think this needs to be resolved before it can be merged

Reply:
host and device signatures should always match exactly

Reply:
They do for `__esimd_dpas*` intrinsics used by `dpas/w` from `math.hpp'. This `__esimd_dpas_inner` is only for host/esimd_emulator.

Reply:
I see, thanks. Still, void* should be avoided as it decreases readability and can be source of bugs.
Also, please roll-back unrelated formatting changes.

Reply:
I reverted this `void*` work-around as the failing test is passing locally.

Reply:
`void*` argument is no longer used and tests are passing (with unrelated failures from 'SYCL / Linux / CUDA LLVM Test Suite')

Reply:
Type casting work-around is removed. Now `dpas.cpp` test fails with following error message.

```
/export/users/dongkyun/github/opensource/intel_llvm/build/bin/../include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp:473:1: note: candidate function template not viable: no known conversion from 'short *__attribute__((ext_vector_type(16)))' to 'const sycl::ext::intel::esimd::detail::vector_type_t<float, 16U> *' (aka 'const cl::sycl::ext::intel::esimd::detail::raw_vector_type<float, 16>::type *') for 1st argument
__esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<RT, SZ> *src0,

```

Command (From Check-in tests)

```
clang --driver-mode=g++  -D_GLIBCXX_ASSERTIONS=1 -DESIMD_XE_HPC -O0 -fsycl -c -Xclang -emit-llvm  $LLVM_SRC/sycl/test/esimd/dpas.cpp -o dpas.cpp.tmp
```